### ISSUE TITLE
LDAP sync for Team Mailboxes

### ISSUE THREAD

**Issue by chibenwa** [linagora/tmail-backend#1115]
## Why?

Automate Team mailboxes management through LDAP.

Be able to synchronise a list of groups from the LDAP as Twake Mail mailboxes.

Those groups could then be defined from a set of rules defined and replicated by a LinID bot.

This approach is very flexible and hopefully not specific to James / mails. We only need a LSC connector

## LDAP data

I want to synchronise `ou=groups,dc=james,dc=org` in the following LDIF:

```
dn: ou=groups, dc=james,dc=org
ou: groups
objectClass: organizationalUnit

dn: cn=sales,ou=groups, dc=james,dc=org
objectclass: top
objectclass: groupofnames
member: uid=usera,ou=people,dc=james,dc=org
member: uid=userb,ou=people,dc=james,dc=org
cn: mygroup
ou: groups

dn: cn=hiring,ou=groups, dc=james,dc=org
objectclass: top
objectclass: groupofnames
member: uid=userc,ou=people,dc=james,dc=org
member: uid=userb,ou=people,dc=james,dc=org
cn: mygroup
ou: groups
```

Resulting in the following team mailboxes:

 - `sales@james.org` with users `usera@james.org` and `userb@james.org`
 - `hiring@james.org` with users `userb@james.org` and `userc@james.org`

Please note that:

 - We can use the CN in order to build the local part of a group.
 - User mails needs to be recovered from their DN and not inferred by their uid
 
## How?

Add a script for LSC twake mail in order to do this.

**Comment by chibenwa** [comment]
@guimard is it possible to attach a `mail` property into the `groupofnames` ? Or we just use the cn for building the team mailbox name?

**Comment by guimard** [comment]
> @guimard is it possible to attach a `mail` property into the `groupofnames` ? Or we just use the cn for building the team mailbox name?

Not directly, we need a schemas change

**Comment by chibenwa** [comment]
Ok then we can likely configure the domain name on LSC, and rely on the cn to build the mail address.

**Comment by guimard** [comment]
> Ok then we can likely configure the domain name on LSC, and rely on the cn to build the mail address.

Not exactly: we need to distinguish groups with mail and groups without. Maybe more, group mailbox can be : 
 - a "team mailbox" : shared messages
 - a mailing list : a copy of mail for each member

**Comment by chibenwa** [comment]
> @chibenwa, @jcabannes, @shepilov: maybe we can use the "businessCategory" here to defined the "role" of the group. 

I am not against but I thought the LDAP base would serve this purpose.

```
dn: ou=tmailbox, dc=james,dc=org
dn: ou=mlist, dc=james,dc=org

dn: cn=sales,ou=tmailbox, dc=james,dc=org
dn: cn=hiring,ou=tmailbox, dc=james,dc=org

dn: cn=paris,ou=mlist, dc=james,dc=org
```

**Comment by guimard** [comment]
> > @chibenwa, @jcabannes, @shepilov: maybe we can use the "businessCategory" here to defined the "role" of the group.
> 
> I am not against but I thought the LDAP base would serve this purpose.
> 
> ```
> dn: ou=tmailbox, dc=james,dc=org
> dn: ou=mlist, dc=james,dc=org
> 
> dn: cn=sales,ou=tmailbox, dc=james,dc=org
> dn: cn=hiring,ou=tmailbox, dc=james,dc=org
> 
> dn: cn=paris,ou=mlist, dc=james,dc=org
> ```
> 
> I think it could be simpler.
> 
> (Can businessCategory be multi-valued?)

But then it exclude having both mail and drive, no ?

**Comment by chibenwa** [comment]
Can businessCategory be multi-valued?

**Comment by chibenwa** [comment]
Ok then I fully agree to use businessCategory

**Comment by jcabannes** [comment]
Ok for me too

**Comment by Arsnael** [comment]
Updated the description of the task with usage of businessCategory for differentiation between team mailbox and ml group.

Let me know if it's ok.

Moving this to ToDo. When can we expect businessCategory put in place on ldap for prod?

**Comment by chibenwa** [comment]
Back to backlog as this is not essential for upcoming client projects.
